pkg/controllers/workgenerator/controller.go

```diff
@@ -757,9 +810,37 @@ func getWorkNamePrefixFromSnapshotName(resourceSnapshot *fleetv1beta1.ClusterRes
 // setBindingStatus sets the binding status based on the works associated with the binding.
 func setBindingStatus(works map[string]*fleetv1beta1.Work, resourceBinding *fleetv1beta1.ClusterResourceBinding) {
 	bindingRef := klog.KObj(resourceBinding)
+
+	// Note (chenyu1): the work generator will refresh the status of a ClusterResourceBinding with
+	// the following logic:
+	//
+	// a) If the currently active apply strategy (as dictated by the ClusterResourceBinding spec)
+	//    is ClientSideApply or ServerSideApply, the work generator will update the Applied and
+	//    Available conditions (plus the details about failed, diffed, and/or drifted placements)
+	//    in the status, as appropriate; the DiffReported condition will not be updated.
+	// b) If the currently active apply strategy is ReportDiff, the work generator will update
+	//    the DiffReported condition in the status, plus the details about diffed placements;
+	//    the Applied condition will always be set to False, despite the fact that the work applier
```

Comment:
where is " the Applied condition will always be set to False" done explicitly?

Reply:
Hi Ryan! When the apply strategy changes, the work object's generation is bumped (all existing conditions become stale). The work applier will not set the Applied condition in the ReportDiff mode, and the `buildAllWorkAppliedCondition` function in the work generator will hence set a false Applied condition

Reply:
Note to self: need to revise the work just updated status handling code a little and add additional test suites if a decision has been made,

Reply:
> The work applier will not set the Applied condition in the ReportDiff mode,

IIRC, That's not really "explicit" but more like "by default".

> This of course would need the rollout controller to processing ReportDiff mode a bit differently, but it feels more like a clean cut

My PR is changing the Rollout controller to handle ReportDiff.

Reply:
Changes made after the latest round of discussions.